WIP global rephrasing (#222)
meilisearch/documentation: guides/main_concepts/indexes.md

# Indexes

> An index is the collection of a certain type of data, an entity that gathers a set of documents.

Author: Yes, much better!

#### Example

Suppose you manage a database that contains information about movies. You would probably want to have multiple categories: one for movie descriptions, one for actors, and one for reviews. Each of these categories would be represented by an index in MeiliSearch.

Each index holds information about the fields found in the documents, how they get handled by MeiliSearch, and their order of importance. Besides, an index defines its own synonyms, relevancy rules, and stop words. The information of one index doesn't impact other indexes.

> For example, it means you could create on the same server synonyms for a `movie` index and other synonyms which would be different for a `clothing` index.

Reviewer: ### 
```suggestion
For example, it means you could create on the same server synonyms for a `movie` index and different synonyms for a `clothing` index.
```

## Index UID

> The `uid` is the **unique** identifier of a given index. It can be found on every `indexes/:index_uid` route as `:index_uid` parameter.

Reviewer: ```suggestion
The `uid` is the **unique** identifier of a given index. It is used on every `indexes/:index_uid` route as the `:index_uid` parameter.
```

## Primary key

> The primary key is a <clientGlossary word="field"/> present in all documents. This field is composed of a primary key <clientGlossary word="attribute"/> name and a unique value. All documents in a given index share the same primary key attribute but a different unique value. The primary key's attribute name **must** be known by the index. You can [set a primary key for an index](/guides/main_concepts/documents.md#setting-the-primary-key) in several ways.

Reviewer: ```suggestion
The primary key is a <clientGlossary word="field"/> present in all documents. This field is composed of a primary key <clientGlossary word="attribute"/> name and a unique value. All documents in a given index share the same primary key attribute but a different unique value. The primary key's attribute name **must** be known by the index. You can [set a primary key for an index or let it be inferred by MeiliSearch](/guides/main_concepts/documents.md#setting-the-primary-key).
```

Author: Thanks! Much better!

[Learn more about document primary key](/guides/main_concepts/documents.md#primary-key)

## Relevancy rules

> Each index applies its own relevancy rules. All indexes are created configured with the same default <clientGlossary word="ranking rules"/> executed in a default order. Once your first document has been added, the index will record how the keys must be sorted. Their order of importance will be deduced from how they were listed in the document.

Reviewer: I'm not sure about the word "Deducted" in the last sentence.
```suggestion
Each index applies its own relevancy rules. All indexes are created with the same built-in <clientGlossary word="ranking rules"/> executed in a default order. Once your first document has been added, the index will record how the attributes must be sorted. Their order of importance will be deducted from their order of appearance in the document.
```

Author: Deducted is more like subtracted (math operation) and deduced means a conclusion was drawn.

Reviewer: Ah yes, I have edited my proposition. Feel free to change it if you don't like it

> For example, if in your first document keys are listed as follows: `id, title, description, release_date`, any document containing the matching query in `title` will be considered more relevant than a document containing it in `description`.

Reviewer: ```suggestion
For example, if in your first document attributes are listed as follows: `id, title, description, release_date`, any document containing the matching query in `title` will be considered more relevant than a document containing it in `description`.
```

On top of that, you can add your custom rules to the ranking rules. For example, you may want to rank your movies either by release date or popularity, or both and so on. Rules are customizable so the results meet your user's needs as close as possible.

[Learn more about ranking rules](/guides/main_concepts/relevancy.md)

## Synonyms

> A set of synonyms can be defined for an index. In your dataset, you may decide to create synonyms for words which have the same meaning. Even though they look different, they should be treated similarly. If either of the associated words is searched, the same results shall be displayed. Since synonyms are linked to a given index, they won't apply to any other index on the same MeiliSearch instance.

Reviewer: ```suggestion
A set of synonyms can be defined for an index. In your dataset, you may decide to create synonyms for words which have the same meaning. Even though they are different, they should be treated similarly. If either of the associated words is searched, the same results shall be displayed. Since synonyms are linked to a given index, they won't apply to any other index on the same MeiliSearch instance.
```

[Learn more about synonyms](/guides/advanced_guides/synonyms.md)

## Stop words

> Sometimes you may want to ignore certain words in documents and search queries. To do so, a set of stop words can be defined for an index. Unless you actually need them, some words neither add semantic value nor context. Besides, they are often too frequent (i.e. `the` or `of` in English).

Reviewer: What a beautiful sentence.

> By adding words to a stop words list, these specific terms will be excluded from your search. It will avoid documents being considered highly relevant because of the high frequency of some words in a corpus.

Reviewer: So i made the same assumption as you just did, but a document containing multiple time a search query will not be considered more relevant than another that has those search query words only once.

But if `The` is present in the `title` (which is the most important attribute) and also at the start of `title` (see position in the ranking rules) the document will be considered highly relevant. 

Furthermore, `the` being so popular, when searching for `the` Meilisearch will have to first gather all documents that contains `the` and then start sorting them. Which could be 90% of your database. Which makes the search slower.
```suggestion
By adding words to a stop words list, these specific terms will be excluded from your search. It will avoid documents being considered highly relevant because of the presence of some words in an important attribute(/guides/main_concepts/relevancy.md#_4-attribute) or in a good [position](/guides/main_concepts/relevancy.md#ranking-rules).
This will also greatly improve the search time because all the documents containing only those words will not be used in the sorting of documents.
```

Author: Oh! Right, I get it (attribute importance)! :o Thanks!
